guides/search.md

```diff
@@ -0,0 +1,147 @@
+                    .Name(o => o.Year)
+                    .Type(NumberType.Integer)))));
+    var movies = new List<Movie>
+    {
+        new Movie { Title = "The Godfather", Director = "Francis Ford Coppola", Year = 1972 },
+        new Movie { Title = "The Shawshank Redemption", Director = "Frank Darabont", Year = 1994 },
+    };
+    for (var i = 0; i < 10; ++i)
+        movies.Add(new Movie { Title = "The Dark Knight " + i, Director = "Christopher Nolan", Year = 2008 + i });
+    // Index the movies
+    var indexResponse = await client.BulkAsync(b => b
+        .Index("movies")
+        .IndexMany(movies)
+        .Refresh(Refresh.WaitFor));
+    Console.WriteLine($"Indexed {indexResponse.Items.Count} movies"); // Indexed 12 movies
+
+    // Refresh the documents to make them searchable
+    var refreshResponse = await client.Indices.RefreshAsync("movies");
+```
+
+
+## Search API
+
+### Basic Search
+The search API allows you to search for documents in an index. The following example searches for ALL documents in the `movies` index:
+
+```csharp
+var searchResponse = await client.SearchAsync<Movie>(s => s
+    .Index("movies")
+    .Query(q => q.MatchAll()));
+Console.WriteLine(string.Join('\n', searchResponse.Documents));
+```
+
+You can also search for documents that match a specific query. The following example searches for documents that match the query `dark knight`:
+```csharp
+var searchResponse = await client.SearchAsync<Movie>(s => s
+    .Index("movies")
+    .Query(q => q
+        .Match(m => m
+            .Field(f => f.Title)
+            .Query("dark knight"))));
+Console.WriteLine(string.Join('\n', searchResponse.Documents));
+```
+
+OpenSearch query DSL allows you to specify complex queries. Check out the [OpenSearch query DSL documentation](https://opensearch.org/docs/latest/query-dsl/) for more information.
+
+
+### Basic Pagination
+The search API allows you to paginate through the search results. The following example searches for documents that match the query `dark knight`, sorted by `year` in in ascending order, and returns the first 2 results after skipping the first 5 results:
+
+```csharp
+var query = Query<Movie>.Match(m => m
+    .Field(f => f.Title)
+    .Query("dark knight"));
+var sort = new SortDescriptor<Movie>()
+    .Ascending(f => f.Year);
+var searchResponse = await client.SearchAsync<Movie>(s => s
+    .Index("movies")
+    .Query(_ => query)
+    .Sort(_ => sort)
+    .From(5)
+    .Size(2));
+Console.WriteLine(string.Join('\n', searchResponse.Documents));
+```
+
+### Sorting
+With sorting, you can also use the `search_after` parameter to paginate through the search results. Let's say you have already displayed the first page of results, and you want to display the next page. You can use the `search_after` parameter to paginate through the search results. The following example will demonstrate how to get the first 3 pages of results using the search query of the previous example:
```

Comment:
This header isn't needed, and `search_after` should be `SearchAfter` to match the code:
```suggestion
With sorting, you can also use the `SearchAfter` parameter to paginate through the search results. Let's say you have already displayed the first page of results, and you want to display the next page. You can use the `SearchAfter` parameter to paginate through the search results. The following example will demonstrate how to get the first 3 pages of results using the search query of the previous example:
```